File: MyProject/C++/NoFila.cpp
#include "NoFila.hpp"

NoFila::NoFila(NoArvore* noArvore)
    : No(noArvore->getFrequencia()), noArvore(noArvore), proximo(nullptr) {}

NoArvore* NoFila::getNoArvore() const { return noArvore; }
void NoFila::setNoArvore(NoArvore* na) { this->noArvore = na; }

NoFila* NoFila::getProximo() const { return proximo; }
void NoFila::setProximo(NoFila* prox) { this->proximo = prox; }

File: MyProject/NoFila.java
public class NoFila extends No {
    
    private NoArvore noArvore;

    private NoFila proximo;

    public NoFila(NoArvore noArvore) {
        super(noArvore.getFrequencia());
        this.noArvore = noArvore;
        this.proximo = null;
    }

    public NoArvore getNoArvore() {
        return noArvore;
    }

    public void setNoArvore(NoArvore noArvore) {
        this.noArvore = noArvore;
    }

    public NoFila getProximo() {
        return proximo;
    }

    public void setProximo(NoFila proximo) {
        this.proximo = proximo;
    }
}